Tests/OptimizelyTests-Common/DataStoreTests.swift

```diff
@@ -97,26 +97,64 @@ class DataStoreTests: XCTestCase {
         // This is an example of a functional test case.
         // Use XCTAssert and related functions to verify your tests produce the correct results.
         
-        let datastore = DataStoreMemory<String>(storeName: "testingDataStoreUserDefaults")
+        let datastore = DataStoreUserDefaults()
         
         datastore.saveItem(forKey: "testString", value: "value")
         
         let value = datastore.getItem(forKey: "testString") as! String
         
         XCTAssert(value == "value")
         
-        datastore.unsubscribe()
+    }
+
+    func testUserDefaultsTooBig() {
+        // This is an example of a functional test case.
+        // Use XCTAssert and related functions to verify your tests produce the correct results.
         
-        datastore.subscribe()
+        let datastore = DataStoreUserDefaults()
         
-        datastore.load(forKey: "testingDataStoreUserDefaults")
+        class Logger : OPTLogger {
+            public var messages: [String]
+                = []
+            required init() {
+                
+            }
+            static var logLevel: OptimizelyLogLevel {
+                set {
+                    
+                }
+                get {
+                    return OptimizelyLogLevel.info
+                }
+            }
+            
+            func log(level: OptimizelyLogLevel, message: String) {
+                messages.append(message)
+            }
+            
+            
+        }
+        let logger = Logger()
         
-        let v2 = datastore.getItem(forKey: "testString") as! String
+        let binder: Binder = Binder<OPTLogger>(service: OPTLogger.self).to { () -> OPTLogger? in
+            return logger
+        }
+
+        HandlerRegistryService.shared.registerBinding(binder: binder)
         
-        XCTAssert(v2 == value)
```

Comment:
Let's clean up here test data which may be left from previous test:
   UserDefaults.standard.removeObject(forKey: "testUserDefaultsTooBig")

Reply:
it is never saved because it is too big.  :)